Sources/SQLTokenizer.swift

```diff
@@ -0,0 +1,303 @@
+import Foundation
+
+public class SQLTokenizer {
+    internal enum Token {
+        case Error
+        case Hexadecimal
+        case Identifier
+        case Integer
+        case Keyword
+        case Literal
+        case Operator
+        case Parameter
+        case Real
+        case Terminal
+        case Whitespace
+        case Other(String)
+        
+        var rawValue: String {
+            switch self {
+            case .Error:
+                return "Error"
+            case .Hexadecimal:
+                return "Hexadecimal"
+            case .Identifier:
+                return "Identifer"
+            case .Integer:
+                return "Integer"
+            case .Keyword:
+                return "Keyword"
+            case .Literal:
+                return "Literal"
+            case .Operator:
+                return "Operator"
+            case .Parameter:
+                return "Parameter"
+            case .Real:
+                return "Real"
+            case .Terminal:
+                return "Terminal"
+            case .Whitespace:
+                return "Whitespace"
+            case .Other(let token):
+                return token
+            }
+        }
+    }
+    
+    public struct TokenValue {
+        let token: Token
```

Comment:
this and `value` need to be marked as public as well if you want them visible to importers (which I think you do?)